src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -103,13 +105,106 @@ struct linux_dirent64 {
   char* d_name;       // Filename (null-terminated)
 };
 
-/** A Linux kernel syscall emulation implementation, which mimics the
-   responses to Linux system calls. */
+/** The types of changes that can be made to values within the process state. */
+enum class ChangeType { REPLACEMENT, INCREMENT, DECREMENT };
+
+/** A structure describing a set of changes to the process state. */
+struct ProcessStateChange {
+  /** Type of changes to be made */
+  ChangeType type;
+  /** Registers to modify */
+  std::vector<Register> modifiedRegisters;
+  /** Values to set modified registers to */
+  std::vector<RegisterValue> modifiedRegisterValues;
+  /** Memory address/width pairs to modify */
+  std::vector<MemoryAccessTarget> memoryAddresses;
+  /** Values to write to memory */
+  std::vector<RegisterValue> memoryAddressValues;
+};
+
+/** The result from a handled syscall. */
+struct SyscallResult {
+  /** Whether execution should halt. */
+  bool fatal;
+  /** Id of the syscall to aid exception handler processing. */
+  uint64_t syscallId;
+  // The unique ID of the core associated with the syscall
+  uint64_t coreId;
+  /** Any changes to apply to the process state. */
+  ProcessStateChange stateChange;
+};
+
+/** A struct to hold information used as arguments to a syscall. */
+struct SyscallInfo {
+  // The ID of the syscall
+  uint64_t syscallId;
+  // The unique sequenceID of the instructions triggering the syscall
+  uint64_t seqId;
+  // The unique ID of the core associated with the syscall
+  uint64_t coreId;
+  // The unique ID of the process associated with the syscall
+  uint64_t processId;
+  // The register values used a parameters to the envoked syscall
```

Comment:
Could these values be stored in std::array or std::vector?

Reply:
Changed to be an `std::array`